core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,137 @@
+        //somewhere else in the future?
+        // Feed dirFeed = DBTasks.updateFeed(context, feed)[0];
+        Feed dirFeed = feed;
+
+        String uriStr = dirFeed.getDownload_url();
+        File f = new File(uriStr.substring("file:".length())); //ugly
+
+        //find relevant files and create items for them
+        File[] itemFiles = f.listFiles(
+                new FilenameFilter() {
+                    @Override
+                    public boolean accept(File file, String s) {
+                        String m = getMimeType(s);
+                        return m != null && (m.startsWith("audio/") || m.startsWith("video/"));
+                    }
+                });
+        List<FeedItem> newItems = dirFeed.getItems();
+        for (File it: itemFiles) {
+            FeedItem found = feedContainsFile(dirFeed, it.getAbsolutePath());
+            if (found != null) {
+                //TODO update (not implemented yet)
+            } else {
+                FeedItem item = createFeedItem(it, dirFeed);
+                newItems.add(item);
+            }
+        }
+
+        //TODO do something about feeditems whose files are no longer in the directory
+
+        //is there an icon? use the first one found
+        File[] iconFiles = f.listFiles(
+                new FilenameFilter() {
+                    @Override
+                    public boolean accept(File file, String s) {
+                        return s.toLowerCase().equals("folder.jpg") ||
+                                s.toLowerCase().equals("folder.png");
+                    }
+                }
+        );
+        if (iconFiles.length > 0) {
+            dirFeed.setImageUrl(
+                    new Uri.Builder().scheme("file").path(iconFiles[0].getAbsolutePath())
+                            .build().toString());
+        }
+
+        //merge to the db
+        Feed[] feeds = DBTasks.updateFeed(context, dirFeed);
+    }
+
+    private static FeedItem feedContainsFile(Feed feed, String fileUrl) {
+        List<FeedItem> items = feed.getItems();
+        for (FeedItem i: items) {
+            if (i.getMedia().getFile_url().equals(fileUrl)) {
+                return i;
+            }
+        }
+        return null;
+    }
+
+    private static FeedItem createFeedItem(File f, Feed whichFeed) {
+        //create item
+        long globalId = 0;
+        Date date = new Date();
+        String uuid = UUID.randomUUID().toString();
+        FeedItem item = new FeedItem(globalId, f.getName(), uuid,
+                f.toString(), date, FeedItem.UNPLAYED, whichFeed);
+        item.setAutoDownload(false);
+
+        //add the media to the item
+        long duration = getFileDuration(f);
+        long size = f.length();
+        String absPath = f.getAbsolutePath();
+        FeedMedia media = new FeedMedia(0, item, (int)duration, 0, size, getMimeType(absPath), absPath, absPath, true, null, 0, 0);
+        item.setMedia(media);
+
+        return item;
+    }
+
```

Comment:
To report errors to the user, you can use something like this:

```
private static void reportError(Context context, Feed feed, String reasonDetailed) {
    DownloadStatus status = new DownloadStatus(-1, Feed.FEEDFILETYPE_FEED, feed.getTitle(),
            DownloadError.ERROR_IO_ERROR, false, reasonDetailed);
    DBWriter.addDownloadStatus(status);
    
    feed.setLastUpdateFailed(true);
    DBTasks.updateFeed(context, feed);
}
````

Reply:
That seems to work, but I can't see any indication of error in the drawer, only in the Downloads->Log list.